src/vs/workbench/contrib/extensions/browser/extensionsActions.ts

```diff
@@ -838,8 +838,20 @@ export class UpdateAction extends ExtensionAction {
 		const isInstalled = this.extension.state === ExtensionState.Installed;
 
 		this.enabled = canInstall && isInstalled && this.extension.outdated;
-		this.class = this.enabled ? UpdateAction.EnabledClass : UpdateAction.DisabledClass;
-		this.label = this.getLabel(this.extension);
+		this.class = this.enabled ? AbstractUpdateAction.EnabledClass : AbstractUpdateAction.DisabledClass;
+		this.label = this.getLabel();
+	}
+
+	protected abstract getLabel(): string;
```

Comment:
This abstract method is not needed. Since labels are static you can pass the labels in the cunstructor.